crates/aptos-metrics/src/system_information.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::system_metrics::SYSTEM;
+use aptos_infallible::Mutex;
+use once_cell::sync::Lazy;
+use std::collections::HashMap;
+use sysinfo::{CpuExt, DiskExt, System, SystemExt};
+
+/// Constants for the keys in the system information map
+const CPU_BRAND: &str = "cpu_brand";
+const CPU_COUNT: &str = "cpu_count";
+const CPU_FREQUENCY: &str = "cpu_frequency";
+const CPU_NAME: &str = "cpu_name";
+const CPU_VENDOR_ID: &str = "cpu_vendor_id";
+const DISK_AVAILABLE_SPACE: &str = "disk_available_space";
+const DISK_COUNT: &str = "disk_count";
+const DISK_FILE_SYSTEM: &str = "disk_file_system";
+const DISK_NAME: &str = "disk_name";
+const DISK_TOTAL_SPACE: &str = "disk_total_space";
+const DISK_TYPE: &str = "disk_type";
+const GIT_REVISION: &str = "git_revision";
+const SYSTEM_HOST_NAME: &str = "system_host_name";
+const SYSTEM_KERNEL_VERSION: &str = "system_kernel_version";
+const SYSTEM_NAME: &str = "system_name";
+const SYSTEM_OS_VERSION: &str = "system_os_version";
+
+/// Used to expose system information.
+pub fn get_system_information() -> HashMap<String, String> {
+    let mut system_information: HashMap<String, String> = HashMap::new();
+    collect_git_revision(&mut system_information);
+    collect_system_info(&mut system_information);
+    system_information
+}
+
+/// Collects the git revision and appends it to the given metric map
+fn collect_git_revision(system_information: &mut HashMap<String, String>) {
+    let git_rev = get_git_rev();
+    if !git_rev.is_empty() {
+        system_information.insert(GIT_REVISION.to_string(), git_rev);
+    }
+}
+
+/// Fetches the git revision from the environment variables
+pub fn get_git_rev() -> String {
+    env!("GIT_REV").to_string()
```

Comment:
I was planning to use `shadow.rs` for this. See how the CLI does it https://github.com/aptos-labs/aptos-core/blob/85425247c7306c741cc73a1d8b6fe8a82060fd42/crates/aptos/src/common/utils.rs#L93

Reply:
Also, I've updated the PR description with an example of fetching the system information on my mac. Mac information isn't super accurate when using the sys info crate. But, I figured an example couldn't hurt 😄

Reply:
Awesome, thanks @rustielin! Yes, for sure. Once I have the schema stable, I'll write up an API document and update the dev docs!